packages/react-native/scripts/cocoapods/privacy_manifest_utils.rb

```diff
@@ -70,13 +70,17 @@ def self.ensure_reference(file_path, user_project, target)
         reference_exists = target.resources_build_phase.files_references.any? { |file_ref| file_ref.path.end_with? "PrivacyInfo.xcprivacy" }
         unless reference_exists
             # We try to find the main group, but if it doesn't exist, we default to adding the file to the project root – both work
-            file_root = user_project.root_object.main_group.children.first { |group| group.name == target.name } || user_project
+            file_root = user_project.root_object.main_group.children.find { |group| group.class == Xcodeproj::Project::Object::PBXGroup && (group.name == target.name || group.path == target.name) } || user_project
             file_ref = file_root.new_file(file_path)
             build_file = target.resources_build_phase.add_file_reference(file_ref, true)
         end
     end
 
     def self.get_privacyinfo_file_path(user_project)
+        existing_file = user_project.files.find { |file_ref| file_ref.path.end_with? "PrivacyInfo.xcprivacy" }
```

Comment:
Not sure about this change, since it could also be a PrivacyInfo file from pods – maybe we should look at the target resource_build_phase files instead?